File: 6-day/rock-paper-scissors/rock-paper-scissors.py
def rps(opt1, opt2):
    return {
        'rock': {
            'rock': 'rock vs rock -> Drawn',
            'paper': 'rock vs paper -> Player 2 win',
            'scissors': 'rock vs scissors -> Player 1 win',
        }.get(opt2, 'Player 2, invalid option'),
        'paper': {
            'rock': 'paper vs rock -> Player 1 win',
            'paper': 'paper vs paper -> Drawn',
            'scissors': 'paper vs scissors -> Player 2 win',
        }.get(opt2, 'Player 2, invalid option'),
        'scissors': {
            'rock': 'scissors vs rock -> Player 2 win',
            'paper': 'scissors vs paper -> Player 1 win',
            'scissors': 'scissors vs scissors -> Drawn',
        }.get(opt2, 'Player 2, invalid option'),
    }.get(opt1, 'Player 1, invalid option')

opt1 = input('Player 1 ¿rock, paper or scissors?: ')
opt2 = input('Player 2 ¿rock, paper or scissors?: ')
print(rps(opt1, opt2))

File: 7-day/rock-paper-scissors.js
function rps(opt1, opt2) {
  switch (opt1) {
    case 'rock':
      switch (opt2) {
        case 'rock': return 'rock vs rock -> Drawn'
        case 'paper': return 'rock vs paper -> Player 2 win'
        case 'scissors': return 'rock vs scissors -> Player 1 win'
        default: return 'Player 2, invalid option'
      }
    case 'paper':
      switch (opt2) {
        case 'rock': return 'paper vs rock -> Player 1 win'
        case 'paper': return 'paper vs paper -> Drawn'
        case 'scissors': return 'paper vs scissors -> Player 2 win'
        default: return 'Player 2, invalid option'
      }
    case 'scissors':
      switch (opt2) {
        case 'rock': return 'scissors vs rock -> Player 2 win'
        case 'paper': return 'scissors vs paper -> Player 1 win'
        case 'scissors': return 'scissors vs scissors -> Drawn'
        default: return 'Player 2, invalid option'
      }
    default:
      return 'Player 1, invalid option'
  }
}

let opt1 = prompt('Player 1: rock, paper or scissors?')
let opt2 = prompt('Player 2: rock, paper or scissors?')
alert(rps(opt1, opt2))